src/code/z_quake.c

```diff
@@ -40,230 +33,256 @@ typedef enum {
-        f32 perpendicularPertubation = Math_SinS(req->speed * req->countdown);
+    if (req->timer > 0) {
+        f32 xyOffset = Math_SinS(req->speed * req->timer);
 
-        Quake_UpdateShakeInfo(req, shake, perpendicularPertubation, perpendicularPertubation);
-        req->countdown--;
+        Quake_UpdateShakeInfo(req, shake, xyOffset, xyOffset);
+        req->timer--;
     }
-    return req->countdown;
+    return req->timer;
 }
 
 s16 Quake_CallbackType6(QuakeRequest* req, ShakeInfo* shake) {
-    f32 perpendicularPertubation;
+    f32 xyOffset;
+
+    req->timer--;
+    xyOffset = Math_SinS(req->speed * ((req->timer & 0xF) + 500));
+    Quake_UpdateShakeInfo(req, shake, xyOffset, Rand_ZeroOne() * xyOffset);
 
-    req->countdown--;
-    perpendicularPertubation = Math_SinS(req->speed * ((req->countdown & 0xF) + 500));
-    Quake_UpdateShakeInfo(req, shake, perpendicularPertubation, Rand_ZeroOne() * perpendicularPertubation);
+    // Not returning the timer ensures quake type 6 continues indefinitely until manually removed
     return 1;
 }
 
 s16 Quake_CallbackType3(QuakeRequest* req, ShakeInfo* shake) {
-    if (req->countdown > 0) {
-        f32 perpendicularPertubation =
-            Math_SinS(req->speed * req->countdown) * ((f32)req->countdown / (f32)req->countdownMax);
+    if (req->timer > 0) {
+        f32 xyOffset = Math_SinS(req->speed * req->timer) * ((f32)req->timer / req->duration);
 
-        Quake_UpdateShakeInfo(req, shake, perpendicularPertubation, perpendicularPertubation);
-        req->countdown--;
+        Quake_UpdateShakeInfo(req, shake, xyOffset, xyOffset);
+        req->timer--;
     }
-    return req->countdown;
+    return req->timer;
 }
 
 s16 Quake_CallbackType2(QuakeRequest* req, ShakeInfo* shake) {
-    if (req->countdown > 0) {
-        f32 perpendicularPertubation = Quake_Random();
+    if (req->timer > 0) {
+        f32 xyOffset = Quake_Random();
 
-        Quake_UpdateShakeInfo(req, shake, perpendicularPertubation, Rand_ZeroOne() * perpendicularPertubation);
-        req->countdown--;
+        Quake_UpdateShakeInfo(req, shake, xyOffset, Rand_ZeroOne() * xyOffset);
+        req->timer--;
     }
-    return req->countdown;
+    return req->timer;
 }
 
 s16 Quake_CallbackType4(QuakeRequest* req, ShakeInfo* shake) {
-    if (req->countdown > 0) {
-        f32 perpendicularPertubation = Quake_Random() * ((f32)req->countdown / (f32)req->countdownMax);
+    if (req->timer > 0) {
+        f32 xyOffset = Quake_Random() * ((f32)req->timer / req->duration);
 
-        Quake_UpdateShakeInfo(req, shake, perpendicularPertubation, Rand_ZeroOne() * perpendicularPertubation);
-        req->countdown--;
+        Quake_UpdateShakeInfo(req, shake, xyOffset, Rand_ZeroOne() * xyOffset);
+        req->timer--;
     }
-    return req->countdown;
+    return req->timer;
 }
 
 s16 Quake_GetFreeIndex(void) {
     s32 i;
-    s32 ret = 0;
-    s32 min = 0x10000;
+    s32 index = 0;
+    s32 timerMin = 0x10000; // UINT16_MAX + 1
```

Comment:
Because it stores the minimum value from `sQuakeRequests[i].timer`, which is a short. So this value is chosen because it's 1 larger than the max number 16 bits can hold

Reply:
Oh so it starts off beyond the range above, okay, with you now.

Reply:
I'll expand this comment to help